src/source/Ice/Network.c

```diff
@@ -291,7 +291,14 @@ STATUS socketConnect(PKvsIpAddress pPeerAddress, INT32 sockfd)
     }
 
     retVal = connect(sockfd, peerSockAddr, addrLen);
-    CHK_ERR(retVal >= 0 || getErrorCode() == EINPROGRESS, STATUS_SOCKET_CONNECT_FAILED, "connect() failed with errno %s",
+    /* Windows uses EWOULDBLOCK (WSAEWOULDBLOCK) to indicate connection attempt
+     * cannot be completed immediately, whereas POSIX uses EINPROGRESS. */
+#ifdef _WIN32
```

Comment:
Consider defining a KVS specific error code - say KVS_SOCKET_IN_PROGRESS in the header file which would then define at compile time to the specific platform status code. Here, it would simply check for the value without conditional compilation. This, making the main code more readable.